src/Facets/GlacisFacet.sol

```diff
@@ -0,0 +1,117 @@
+    IGlacisAirlift public immutable airlift;
+
+    /// Types ///
+
+    /// @param refundAddress The address that would receive potential refunds on destination chain
+    /// @param nativeFee The fee amount in native token required by the Glacis Airlift
+    struct GlacisData {
+        address refundAddress;
+        uint256 nativeFee;
+    }
+
+    /// Constructor ///
+    /// @notice Initializes the GlacisFacet contract
+    /// @param _airlift The address of Glacis Airlift contract.
+    constructor(IGlacisAirlift _airlift) {
+        airlift = _airlift;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Glacis
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _glacisData Data specific to Glacis
+    function startBridgeTokensViaGlacis(
+        ILiFi.BridgeData memory _bridgeData,
+        GlacisData calldata _glacisData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _glacisData);
+    }
+
+    /// @notice Performs a swap before bridging via Glacis
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _glacisData Data specific to Glacis
+    function swapAndStartBridgeTokensViaGlacis(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        GlacisData calldata _glacisData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            _glacisData.nativeFee
+        );
+        _startBridge(_bridgeData, _glacisData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Glacis
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _glacisData Data specific to Glacis
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        GlacisData calldata _glacisData
+    ) internal {
+        // Transfer the tokens to the Airlift contract.
+        // This step ensures that the tokens are already in place before calling the `send` function.
+        // The `send` function assumes the tokens are pre-transferred to the contract.
+        SafeERC20.safeTransfer(
```

Comment:
We plan to support only the `sendAfterApproval` method for better gas optimization and security. Could you test this approach to ensure compatibility with LiFi's diamond operations?

Reply:
More specifically, we are replacing the "send" function's functionality/workflow with the "sendAfterApproval" functionality, but it will still be called "send"